### ISSUE TITLE
Enforce empty line separators between different YARD tag groups

### ISSUE THREAD

**Issue by nijikon** [mensfeld/yard-lint#29]
## Summary

Add a new check to detect when different types of YARD tags are not separated by an empty comment line, improving documentation readability and visual organization.

## Current Behavior

yard-lint currently does not enforce separation between different tag types. The following passes without warnings:

```ruby
# @param organization_id [String]
# @param id [String]
# @return [Pet]
def call(organization_id, id)
```

## Proposed Behavior

yard-lint should detect and report when different tag groups are not separated by an empty comment line. The expected clean format would be:

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]
def call(organization_id, id)
```

## Rationale

- **Readability**: Visual separation between tag groups makes documentation easier to scan
- **Logical grouping**: Different tag types serve different purposes (inputs vs outputs vs exceptions) and should be visually distinct
- **Consistency**: Enforces a uniform documentation style across a codebase
- **Common convention**: Many Ruby style guides recommend grouping related documentation elements

### New error type

- `MissingTagGroupSeparator` - For missing empty line between different tag types

### Tag grouping logic

Tags should be grouped by their type/purpose. Common groupings:

1. **Description block** (free text before tags)
2. **Parameter tags**: `@param`, `@option`
3. **Return tags**: `@return`
4. **Error tags**: `@raise`, `@throws`
5. **Example tags**: `@example`
6. **Other meta tags**: `@see`, `@note`, `@todo`, `@deprecated`, `@since`, `@version`, `@api`
7. **Yield tags**: `@yield`, `@yieldparam`, `@yieldreturn`

### Configuration (optional)

Could be configurable to define custom groupings or disable specific separations:

```yaml
# .yard-lint.yml
MissingTagGroupSeparator:
  enabled: true
  # Optionally define which transitions require separation
  require_separation_between:
    - [param, return]
    - [param, raise]
    - [return, raise]
    - [return, example]
```

### Should warn (no separation between @param and @return)

```ruby
# @param id [String]
# @return [Pet]
def find(id)
```

### Should warn (no separation between @return and @raise)

```ruby
# @param id [String]
#
# @return [Pet]
# @raise [NotFoundError]
def find(id)
```

### Should NOT warn (same tag types grouped together)

```ruby
# @param organization_id [String]
# @param id [String]
#
# @return [Pet]
def call(organization_id, id)
```

### Should NOT warn (proper separation between all groups)

```ruby
# Finds a pet by ID within an organization.
#
# @param organization_id [String] the organization identifier
# @param id [String] the pet identifier
#
# @return [Pet] the found pet
#
# @raise [NotFoundError] if pet does not exist
def call(organization_id, id)
```

### Should NOT warn (description separated from tags)

```ruby
# Finds a pet by ID.
#
# @param id [String]
#
# @return [Pet]
def find(id)
```

## Edge cases to consider

- Tags with the same prefix should stay together (e.g., `@yieldparam` and `@yieldreturn` could be in the same group as `@yield`)
- `@option` tags typically follow their associated `@param` and should be in the same group
- Multiple `@example` blocks might warrant separation between them (or not - configurable)
- Custom tags defined by users should have configurable grouping

## Example of fully documented method

```ruby
# Finds a pet by ID within an organization.
#
# This method searches the database for a pet matching the given
# organization and pet IDs.
#
# @param organization_id [String] the organization identifier
# @param id [String] the pet identifier
# @param options [Hash] additional options
# @option options [Boolean] :include_deleted whether to include soft-deleted pets
#
# @return [Pet] the found pet
#
# @raise [NotFoundError] if pet does not exist
# @raise [UnauthorizedError] if access is denied
#
# @example
#   pet = Pets::Find.call("org-123", "pet-456")
#
# @see Pets::List
def call(organization_id, id, options = {})
```

## Related

This enhancement complements existing YARD documentation best practices and could work alongside tag ordering checks if those exist or are planned.
